File: 0746-MinCostClimbingStairs/MinCostClimbingStairs.go
package mincostclimbingstairs

func minCostClimbingStairs(cost []int) int {
	totalCost := make([]int, len(cost)+1)
	for i := 2; i < len(totalCost); i++ {
		totalCost[i] = getMin(totalCost[i-1]+cost[i-1], totalCost[i-2]+cost[i-2])
	}
	return totalCost[len(cost)]
}

func getMin(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: 0746-MinCostClimbingStairs/MinCostClimbingStairs.java

class MinCostClimbingStairs {
    class Solution {
        public int minCostClimbingStairs(int[] cost) {
            int[] totalCost = new int[cost.length + 1];
            for(int i = 2; i < totalCost.length; i++){
                totalCost[i] = Math.min(totalCost[i - 1] + cost[i - 1], totalCost[i - 2] + cost[i - 2]);
            }
            
            return totalCost[cost.length];
        }
    }
}
